src/commoncode/fileset.py

```diff
@@ -186,9 +187,12 @@ def load(location):
     fn = os.path.abspath(os.path.normpath(os.path.expanduser(location)))
     msg = ('File %(location)s does not exist or not a file.') % locals()
     assert (os.path.exists(fn) and os.path.isfile(fn)), msg
-    with open(fn, 'rb') as f:
-        return [l.strip() for l in f if l and l.strip()]
-
+    if on_linux and py2:
+        with open(fn, 'rb') as f:
+           return [l.strip() for l in f if l and l.strip()]
+    else:
+        with open(fn, 'r') as f:
+           return [l.strip() for l in f if l and l.strip()]
```

Comment:
To avoid code duplication you could write instead:
```
    mode = 'rb' if on_linux and py2 else 'r'
    with open(fn, mode) as f:
       return [l.strip() for l in f if l and l.strip()]
```